### DISCUSSION TITLE
Preparation of measurement system failed when running pulseq

### DISCUSSION THREAD

**Discussion by siluhan** [pulseq/pulseq#78] [Q&A] [answered]
Hi there,

We recently installed the latest version of Pulseq on a Siemens XA30 and attempted to execute a few sequences. Initially, the sequences ran successfully for a few iterations. However, an error message indicating “measurement system has failed” appeared during the execution of subsequent sequences.

To resolve this issue, we attempted to reboot the system, and to our surprise, the same sequences ran successfully when we re-ran them. Nevertheless, when we simply repeated running the same sequence, we encountered inconsistent results. Sometimes, the sequence would run successfully, while at other times, it would fail with the same error message.

We would greatly appreciate any assistance you may provide in resolving this problem.

Thank you for your attention to this matter.

Best,
Silu

**Comment by QingpingChen1201** [comment, marked as answer]
Hi Silu,
Happy New Year!
Many thanks for your interest in using Pulseq. Did you try to disable the "libBalance / Grad health" in the Special Card of the user interface? The libBalance can provide additional safety checks for the gradient sub-system. In case your sequence is relatively long and heavy, the libBalance may result in unpredictable measurement failures.
If the error is still there after you disable the "libBalance" option, please feel free to send your .seq file to us for a check via our Pulseq team email address: pulseq.mr@uniklinik-freiburg.de.
I wish you a fruitful new year of 2025 for your research work in using Pulseq:)
Best regards,
Qingping and the Pulseq team

**Reply by siluhan** [reply]
Hi Qingping,
Thank you so much for your prompt reply! I’ll try to disable the “libBalance / Grad health” option to see if that resolves the issue.

However, I’m just curious about the reason behind the error message. When I initially scanned one sequence, it worked successfully. However, when I simply repeated the scan (with the same sequence), the “measurement system has failed” error message appeared (sometimes, it appeared immediately after the first scan, while at other times, it appeared after multiple scans.). Does this indicate that the safety checks performed during each scan are different, even when using the same sequence?

Best,
Silu

**Comment by QingpingChen1201** [comment]
Hi Silu,
we have no access to the source code of libBalance package so the inner life of libBalance is not really clear to us. Based on my understanding, libBalance will stop the measurement if the current measurement or previous measurements are very long and dense (e.g. EPI scans, diffusion scans) to avoid overloading the gradient sub-system. One of our colleagues reported that she can run a diffusion sequence but when she run three same sequences together, the measurement failed and throwed an error very similar to yours. Therefore, I suspect it is libBalance that causes your measurement failures. libBalance provides additional safety checks for your Pulseq-based sequences. After turning it off, your Pulseq-based sequences still undergo safety checks similar as other product sequences.
However, please keep this in mind: it is always YOUR responsibility to ensure the safety of your scanner hardware and volunteers/patients. We provides several tools for safety checks, such as [SAR](https://github.com/pulseq/pulseq/discussions/59), [mechanical resonance](https://github.com/pulseq/pulseq/blob/master/matlab/demoUnsorted/gradSpectrum.m), and [PNS](https://github.com/pulseq/pulseq/blob/master/matlab/demoUnsorted/pnsTest.m).
Best regards,
Qingping and the Pulseq team

**Comment by pulseq-admin** [comment]
Dear Silu, 

Just a comment: I believe libBalance has no "memory" between scans, it only
concerns the current scan. However, the RF supervision system (the one
ensuring SAR safety) has such memory. 

It is also possible that there are some bugs in the Pulseq interpreter (like
memory leaks), which make repeated scans unstable. We are not aware of any
reproducible events of this kind, but there are hints that some problems may
be present in the code. 

Best regards, 
Maxim & the Pulseq team
